charts/kargo/templates/api/cluster-role.yaml

```diff
@@ -19,6 +19,28 @@ rules:
       - patch
       - update
       - delete
+  - apiGroups:
+      - ""
+    resources:
+      - serviceaccounts
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - users
+      - groups
+      - serviceaccounts
+    verbs:
+      - impersonate
```

Comment:
This is a lot of privilege that I know @jessesuen and I both wanted to avoid.

I'll review the rest of the PR and then circle back to here... I think there are ways we can avoid this.

Reply:
Ok... I see what's going on.

I see you are impersonating service accounts to perform a selfsubjectaccess review.

It would be safer to _not_ grant impersonation privileges and instead grant prividleges to conduct a subjectaccessreview (on others; _not_ "self").